app/controllers/api/v1/allocation_events_controller.rb

```diff
@@ -23,38 +35,69 @@ def create
 
     private
 
-      PERMITTED_EVENT_PARAMS = [
+      # TODO: remove constant when allocation `events` endpoint is no longer in use
+      DEPRECATED_EVENT_PARAMS = [
         :type,
         attributes: %i[timestamp event_name],
       ].freeze
 
-      def validate_params
-        AllocationEvents::ParamsValidator.new(event_params).validate!(action_name.to_sym)
+      PERMITTED_CANCEL_PARAMS = [
+        :type,
+        attributes: %i[timestamp cancellation_reason cancellation_reason_comment],
+      ].freeze
+
+      def validate_params!(event_params)
+        AllocationEvents::ParamsValidator.new(event_params).validate!(:cancel)
+      end
+
+      def cancel_params
+        @cancel_params ||= params.require(:data).permit(PERMITTED_CANCEL_PARAMS).to_h
+      end
+
+      def cancel_move_params
+        cancel_params.tap do |params|
+          params[:attributes][:cancellation_reason] = Move::CANCELLATION_REASON_OTHER
```

Comment:
Open to better suggestions for this - there's a rule that says we should always provide the reason `other` for the cancelled underlying moves regardless of the reason chosen for cancelling the allocation.

Reply:
maybe add a comment to record that fact:  `# NB: we should always provide the reason other for the cancelled underlying moves regardless of the reason chosen for cancelling the allocation`

Reply:
Ah I wasn't aware of this, this is new. I thought initially we would mirror all the reasons defined on the allocation cancellations here (and add the ones that don't conform with the defined moves cancellation reasons), unless Other was selected on allocation where we would add the comment.

Reply:
Discussed in person - unfortunately moves and allocations have a different set of cancellation reasons, so for simplicity the move cancellation reason is always set to `other`. We could explore a mapping approach or try to standardise the cancellation reasons across moves and allocations in future.